qfluentwidgets/components/widgets/stacked_widget.py

```diff
@@ -9,50 +9,56 @@
 class OpacityAniStackedWidget(QStackedWidget):
     """ Stacked widget with fade in and fade out animation """
 
-    def __init__(self, parent=None):
+    def __init__(self, parent=None) -> None:
         super().__init__(parent=parent)
-        self.__nextIndex = 0
-        self.__effects = []  # type:List[QPropertyAnimation]
-        self.__anis = []     # type:List[QPropertyAnimation]
-
-    def addWidget(self, w: QWidget):
-        super().addWidget(w)
-
-        effect = QGraphicsOpacityEffect(self)
-        effect.setOpacity(1)
-        ani = QPropertyAnimation(effect, b'opacity', self)
-        ani.setDuration(220)
-        ani.finished.connect(self.__onAniFinished)
-        self.__anis.append(ani)
-        self.__effects.append(effect)
-        w.setGraphicsEffect(effect)
-
-    def setCurrentIndex(self, index: int):
-        index_ = self.currentIndex()
-        if index == index_:
-            return
-
-        if index > index_:
-            ani = self.__anis[index]
-            ani.setStartValue(0)
-            ani.setEndValue(1)
-            super().setCurrentIndex(index)
-        else:
-            ani = self.__anis[index_]
-            ani.setStartValue(1)
-            ani.setEndValue(0)
-
-        self.widget(index_).show()
-        self.__nextIndex = index
-        ani.start()
-
-    def setCurrentWidget(self, w: QWidget):
+        self.__create_animations()
+
+    def setCurrentIndex(self, index: int) -> None:
+        if index == self.currentIndex(): return
+        if not self.widget(index): return # avoids going to nonexisting index
+        
+        # Current index hides, target index shows
+        self.currentWidget().setGraphicsEffect(self._opacity1)
+        self.widget(index).setGraphicsEffect(self._opacity2)
+
+        # Show target index (currently invisible)
+        self.widget(index).show()
+
+        # Start animations
+        self._opacityUp.finished.connect(
+            lambda: self.rst_effects(self.currentWidget(), self.widget(index)))
+        self._opacityDown.start(QAbstractAnimation.DeletionPolicy.KeepWhenStopped)
+        self._opacityUp.start(QAbstractAnimation.DeletionPolicy.KeepWhenStopped)
+
+    def rst_effects(self, w_hidden: QWidget, w_shown: QWidget) -> None:
+        super().setCurrentWidget(w_shown)
+        w_hidden.setGraphicsEffect(None)
+        w_shown.setGraphicsEffect(None)
+        self.__create_animations()
+
+    def __create_animations(self) -> None:
+
+        self._opacity1 = QGraphicsOpacityEffect(self)
+        self._opacity2 = QGraphicsOpacityEffect(self)
+        self._opacity2.setOpacity(0.0)
```

Comment:
`_opacity1` is assigned to former widget whilst `_opacity2` is assigned to widget that is to be shown